pkg/cloud/services/elb/loadbalancer_test.go

```diff
@@ -1000,6 +1003,109 @@ func TestChunkELBs(t *testing.T) {
 	}
 }
 
+func TestInstancePortCustomizationsInELB(t *testing.T) {
+	tests := []struct {
+		name   string
+		kcp    *controlplanev1.KubeadmControlPlane
+		mocks  func(m *mock_ec2iface.MockEC2APIMockRecorder)
+		expect func(t *testing.T, res *infrav1.ClassicELB)
+	}{
+		{
+			name: "load balancer config with kcp bind port customization",
+			kcp: &controlplanev1.KubeadmControlPlane{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cp",
+					Namespace: "foo",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "KubeadmControlPlane",
+					APIVersion: "kubeadm.k8s.io/v1beta2",
+				},
+				Spec: controlplanev1.KubeadmControlPlaneSpec{
+					KubeadmConfigSpec: cabpkv1.KubeadmConfigSpec{
+						InitConfiguration: &cabpkv1.InitConfiguration{
+							LocalAPIEndpoint: cabpkv1.APIEndpoint{
+								BindPort: 443,
+							},
+						},
+					},
+				},
+			},
+			mocks: func(m *mock_ec2iface.MockEC2APIMockRecorder) {},
+			expect: func(t *testing.T, res *infrav1.ClassicELB) {
+				if !res.Attributes.CrossZoneLoadBalancing {
+					t.Error("Expected load balancer to have cross-zone load balancing enabled")
+				}
+			},
+		},
+		{
+			name:  "load balancer config with no kcp bind port customization",
+			kcp:   &controlplanev1.KubeadmControlPlane{},
+			mocks: func(m *mock_ec2iface.MockEC2APIMockRecorder) {},
+			expect: func(t *testing.T, res *infrav1.ClassicELB) {
+				if !res.Attributes.CrossZoneLoadBalancing {
+					t.Error("Expected load balancer to have cross-zone load balancing enabled")
+				}
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+			ec2Mock := mock_ec2iface.NewMockEC2API(mockCtrl)
+
+			scheme, err := setupScheme()
+			if err != nil {
+				t.Fatal(err)
+			}
+			client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(tc.kcp).Build()
+			clusterScope, err := scope.NewClusterScope(scope.ClusterScopeParams{
+				Client: client,
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "foo",
+						Name:      "bar",
+					},
+					Spec: clusterv1.ClusterSpec{
+						ControlPlaneRef: &corev1.ObjectReference{
+							Name:      "test-cp",
+							Kind:      "KubeadmControlPlane",
+							Namespace: "foo",
+						},
+					},
+				},
+				AWSCluster: &infrav1.AWSCluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "test"},
+					Spec: infrav1.AWSClusterSpec{
+						ControlPlaneLoadBalancer: &infrav1.AWSLoadBalancerSpec{
+							CrossZoneLoadBalancing: true,
+						},
+					},
+				},
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			tc.mocks(ec2Mock.EXPECT())
+
+			s := &Service{
+				scope:     clusterScope,
+				EC2Client: ec2Mock,
+			}
+
+			spec, err := s.getAPIServerClassicELBSpec(clusterScope.Name())
```

Comment:
Where do you check if the correct ELB internal port is used?

Reply:
Added the check now